### PR TITLE
PUBDEV-8340: optional SEs are ignored in AutoML

### PR DESCRIPTION
**Pull request by sebhrusen**
https://h2oai.atlassian.net/browse/PUBDEV-8340

Improved recognition of dopplegangers among SE models, especially after introduction of several new SEs in `3.34.0.1`.

### CODE REVIEW

File: h2o-automl/src/main/java/ai/h2o/automl/modeling/StackedEnsembleStepsProvider.java [hunk 1]
```diff
@@ -376,6 +390,11 @@ protected void setMetalearnerParameters(StackedEnsembleParameters params) {
                         }
                     });
                     optionalSeSteps.add(new AllSEModelStep("all_xglm", optionalGroup, aml()) {
+                        @Override
+                        protected boolean hasDoppelganger(Key<Model>[] keys) {
+                            return false;
```
Comment (tomasfryda) [inline h2o-automl/src/main/java/ai/h2o/automl/modeling/StackedEnsembleStepsProvider.java:395]: Can't we do better? Something like `number of base models > number of "best of family" models`. So we don't end up with `all_xglm == best_of_family_xglm`.

File: h2o-py/tests/testdir_algos/automl/pyunit_automl_args.py [hunk 2]
```diff
@@ -417,31 +429,32 @@ def test_default_max_runtime_if_no_max_models_provided():
...
+    # test_automl_stops_after_max_models,
+    # test_stacked_ensembles_are_trained_after_max_models,
+    # test_stacked_ensembles_are_trained_with_blending_frame_even_if_nfolds_eq_0,
+    # test_frames_can_be_passed_as_key,
+    # test_no_time_limit_if_max_models_is_provided,
+    test_optional_SEs_trained_by_default_when_no_time_limit,
+    # test_max_runtime_secs_alone,
+    # test_max_runtime_secs_can_be_set_in_combination_with_max_models_and_max_models_wins,
+    # test_max_runtime_secs_can_be_set_in_combination_with_max_models_and_max_runtime_wins,
+    # test_default_max_runtime_if_no_max_models_provided,
```
Comment (tomasfryda) [inline h2o-py/tests/testdir_algos/automl/pyunit_automl_args.py:459]: These will get uncommented before merging, right?

Comment (sebhrusen) [reply h2o-py/tests/testdir_algos/automl/pyunit_automl_args.py:459]: oups! 2nd time I do it on this PR I think

File: h2o-automl/src/main/java/ai/h2o/automl/modeling/StackedEnsembleStepsProvider.java (hunk 1 above)
Comment (sebhrusen) [reply h2o-automl/src/main/java/ai/h2o/automl/modeling/StackedEnsembleStepsProvider.java:395]: good point, will see how I can improve this: wanted to keep the logic simple for this one.

**Review by tomasfryda — APPROVED** [review]
Looks good to me. Thank you @sebhrusen!

**Review by ledell — APPROVED** [review]
Thanks for the fix, @sebhrusen!

**Comment by ledell** [comment]
@sebhrusen Do you want me to do the follow-up PR for the docs FAQ edit?  Or were you going to do it?

**Pull request merged**